lib/simp/cli/environment/omni_env_controller.rb

```diff
@@ -0,0 +1,84 @@
+require 'simp/cli/environment/env'
+require 'simp/cli/environment/puppet_dir_env'
+require 'simp/cli/environment/secondary_dir_env'
+require 'simp/cli/environment/writable_dir_env'
+
+# Puppetfile helper namespace
+module Simp::Cli::Environment
+  # Controller class to manage SIMP Omni environments
+  class OmniEnvController
+    def initialize(opts = {}, env = nil)
+      @opts = opts
+      @environments = {}
+      @opts[:types].each do |type, data|
+        # TODO: honor backends
+        # TODO: refactor into a Factory
+        base_env_path = data[:environmentpath] || fail(ArgumentError, 'ERROR: no :environmentpath in opts')
+        case type
+        when :puppet
+          @environments[:puppet]    = PuppetDirEnv.new(env, base_env_path, data)
+        when :secondary
+          @environments[:secondary] = SecondaryDirEnv.new(env, base_env_path, data)
+        when :writable
+          @environments[:writable]  = WritableDirEnv.new(env, base_env_path, data)
+        else
+          fail( "ERROR: Unrecognized environment type '#{env_type}'" )
+        end
+      end
+    end
+
+    # Create a new environment for each environment type
+    def create
+      each_environment do |env_type, env_obj|
+        env_obj.create
+      end
+
+      # ensure environment is correct after creating it
+      fix
+    end
+
+    # Update environment
+    def update
+      fail NotImplementedError
+    end
+
+    # Remove environment
+    def remove
+      fail NotImplementedError
+    end
+
+    # List current environments
+    def list
```

Comment:
``simp config`` will need this to return some sort of status object,  ``simp config`` may want to request this information about a specific environment (puppet, secondary, writable).

Reply:
:ballot_box_with_check: